tools/main.py

```diff
@@ -24,31 +25,42 @@ def build_datamanager(cfg, classification_classes_filter=None):
     else:
         return torchreid.data.VideoDataManager(**videodata_kwargs(cfg))
 
-def build_auxiliary_model(config_file, num_classes, use_gpu, device_ids=None, weights=None):
-    cfg = get_default_config()
-    cfg.use_gpu = use_gpu
-    cfg.merge_from_file(config_file)
+def build_auxiliary_model(config_file, num_classes, use_gpu, device_ids=None, lr=None):
+    def compute_s(num_class: int):
+        return float(max(np.sqrt(2) * np.log(num_class - 1), 3))
 
-    model = torchreid.models.build_model(**model_kwargs(cfg, num_classes))
+    aux_cfg = get_default_config()
+    aux_cfg.use_gpu = use_gpu
+    aux_cfg.merge_from_file(config_file)
 
-    if (weights is not None) and (check_isfile(weights)):
-        load_pretrained_weights(model, weights)
+    model = torchreid.models.build_model(**model_kwargs(aux_cfg, num_classes))
 
-    if cfg.use_gpu:
+    if aux_cfg.use_gpu:
         assert device_ids is not None
 
         model = DataParallel(model, device_ids=device_ids, output_device=0).cuda(device_ids[0])
+    if lr is not None:
+        aux_cfg.train.lr = lr
+        print(f"setting learning rate from main model, estimated by lr finder: {lr}")
+    if aux_cfg.loss.name == 'am_softmax':
+        s = compute_s(num_classes[0])
```

Comment:
I think we shouldn't hack config right in builder and instead introduce a parameter in config.